.github/workflows/publish-images.yaml

```diff
@@ -0,0 +1,118 @@
+name: Publish images
+
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
+jobs:
+  prepare:
+    name: Prepare properties
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Prepare build parameters
+        id: prep
+        run: |
+          hack/build/ci/prepare-build-variables.sh
+    outputs:
+      labels: ${{ steps.prep.outputs.docker_image_labels }}
+      version: ${{ steps.prep.outputs.docker_image_tag }}
+      goargs: ${{ steps.prep.outputs.go_build_args }}
+      goversion: ${{ steps.prepenv.outputs.goversion }}
+
+  # Workflow needs to build docker images again as separate workflows don't have access to others artifacts
+  # https://github.com/actions/download-artifact/issues/3
+  build:
+    name: Build images
+    needs: [prepare]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [amd64, arm64]
+    steps:
+      - name: Build image for ${{matrix.platform}}
+        uses: ./.github/workflows/build-dockerimage.yaml
+        with:
+          platform: ${{matrix.platform}}
+          goversion: ${{ needs.prepare.outputs.goversion }}
+          goargs: ${{ needs.prepare.outputs.goargs }}
+          labels: ${{ needs.prepare.outputs.labels }}
+          image-tag: ${{ needs.prepare.outputs.version }}
+  push:
+    name: Push images
+    environment: Release
+    needs: [prepare,build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
I'd love to not duplicate the matrix...
since we always want to do a multi arch build here can we simplify some logic by moving it into the `upload-dockerimage` action?